controllers/utils/common.go

```diff
@@ -85,3 +88,60 @@ func ContainsTemplates(s string) bool {
 
 	return regexpAllTemplates.MatchString(s)
 }
+
+// Difference returns the elements in `a` that aren't in `b`.
+func Difference(a, b []string) []string {
+	mb := make(map[string]struct{}, len(b))
+	for _, x := range b {
+		mb[x] = struct{}{}
+	}
+	var diff []string
+	for _, x := range a {
+		if _, found := mb[x]; !found {
+			diff = append(diff, x)
+		}
+	}
+	return diff
+}
+
+// SortCGUListByIBUAction orders the CGUs by the last action in the name of cgu
+// with following order prep-upgrade-rollback-finalize-abort
+func SortCGUListByIBUAction(cguList *ranv1alpha1.ClusterGroupUpgradeList) {
```

Comment:
This will need to be updated when we support partial abort/rollback in the same CR